### PR TITLE
Add additional information about firmware update

### PR DESCRIPTION
**Pull request by raghavag**
This patch added additional information about firmware update
namely, capsule layout, generating capsule, triggering firmware
update and generating sbl binaries for firmware update capsule.

Signed-off-by: Raghava Gudla <raghava.gudla@intel.com>

### CODE REVIEW

**Comment by raghavag** [comment]
[Whiskey Lake CRB Board — Slim Bootloader 1.0 documentation.pdf](https://github.com/slimbootloader/documentation/files/3799131/Whiskey.Lake.CRB.Board.Slim.Bootloader.1.0.documentation.pdf)
[Apollo Lake CRB Boards — Slim Bootloader 1.0 documentation.pdf](https://github.com/slimbootloader/documentation/files/3799132/Apollo.Lake.CRB.Boards.Slim.Bootloader.1.0.documentation.pdf)
[Firmware Update — Slim Bootloader 1.0 documentation.pdf](https://github.com/slimbootloader/documentation/files/3799133/Firmware.Update.Slim.Bootloader.1.0.documentation.pdf)
[UP2 Board — Slim Bootloader 1.0 documentation.pdf](https://github.com/slimbootloader/documentation/files/3799134/UP2.Board.Slim.Bootloader.1.0.documentation.pdf)

**Comment by raghavag** [comment]
Attaching zip files of html, PDF does not show the alignment properly.

[fwupdate_doc_update_html.zip](https://github.com/slimbootloader/documentation/files/3799140/fwupdate_doc_update_html.zip)

File: source/security/firmware-update.rst [hunk 1]
```diff
@@ -1,37 +1,372 @@
 Firmware Update
 ------------------
 
-|SPN| implements a robust, secure and power fail-safe firmware update mechanism. |SPN| image can be built to contain redundant boot components and depending on hardware assisted boot partition switch, to support fail-safe update.
+|SPN| implements a secure and power fail-safe firmware update mechanism.
 
-If firmware update is interrupted by a power failure and one boot partition is corrupted, |SPN| boots from the alternate partition to avoid bricking the platform. The firmware update authenticates the update image (a.k.a. capsule) before updating the firmware.
+  * |SPN| image is designed to contain redundant boot components and depends on hardware assisted boot partition switch to support fail-safe update.
```
Comment (rprangar) [inline source/security/firmware-update.rst]: Add a couple of points here
* Firmware update code is implemented as a payload
* |SPN| detects firmwate update signal and launches the Firmware Update payload

File: source/security/firmware-update.rst [hunk 2]
```diff
@@ -1,37 +1,372 @@
...
 ------------------
 
-|SPN| implements a robust, secure and power fail-safe firmware update mechanism. |SPN| image can be built to contain redundant boot components and depending on hardware assisted boot partition switch, to support fail-safe update.
+|SPN| implements a secure and power fail-safe firmware update mechanism.
 
-If firmware update is interrupted by a power failure and one boot partition is corrupted, |SPN| boots from the alternate partition to avoid bricking the platform. The firmware update authenticates the update image (a.k.a. capsule) before updating the firmware.
+  * |SPN| image is designed to contain redundant boot components and depends on hardware assisted boot partition switch to support fail-safe update.
 
-Firmware update feature is implemented as a |SPN| payload.
+  * Firmware update code will authenticates the capsule image before proceeding with firmware update.
```
Comment (rprangar) [inline source/security/firmware-update.rst]: authenticate

File: source/security/firmware-update.rst [hunk 3]
```diff
@@ -1,37 +1,372 @@
...
 
-When firmware update is signaled, Stage 2 loads firmware update payload to start update flow. During the update, firmware update payload keeps a state machine to keep track of the update progress as follows:
+  * If firmware update is interrupted by a power failure and a boot partition is corrupted, |SPN| will boot from the alternate partition to avoid bricking the platform.
```
Comment (rprangar) [inline source/security/firmware-update.rst]: This line is not accurate.

File: source/security/firmware-update.rst [hunk 4]
```diff
@@ -1,37 +1,372 @@
...
-#. If update flow is requested, load FWU payload
-#. FWU loads and verifies capsule data. If successful, update partition B
-#. Set partition B as 'active'
-#. Reboot
-#. Boot from partition B
-#. If update flow is in progress, load FWU payload
-#. FWU loads and verifies capsule data. If successful, update partition A
-#. Set partition A as 'active'
-#. End update flow state machine and reboot
+|SPN| supports independent update of BIOS and CSME regions of IFWI. Update of external configuration data is also supported.
```
Comment (rprangar) [inline source/security/firmware-update.rst]: An overview of the high level flow is provided below

File: source/security/firmware-update.rst [hunk 5]
```diff
@@ -1,37 +1,372 @@
...
 
+  **Step 1:**  User triggers firmware update from |SPN| shell or from Operating System and reset the system
```
Comment (rprangar) [inline source/security/firmware-update.rst]: Step 1: Firmware Update capsule is copied to the designated location. This location is configurable through SBL configuration options.
Step 2: Firmware update is signalled from |SPN| shell or from Operating System and the system is reset

File: source/security/firmware-update.rst [hunk 6]
```diff
@@ -1,37 +1,372 @@
...
 
+  **Step 2:**  |SPN| detects firmware update signal and sets platform into firmware update mode.
+
+  **Step 3:**  Stage 2 code identifies firmware update mode and loads firmware update payload to start update flow.
+
+  **Step 4:**  FWU gathers capsule image from selected media and verifies capsule data. If successful, continue with firmware update
```
Comment (rprangar) [inline source/security/firmware-update.rst]: What is FWU?

File: source/security/firmware-update.rst [hunk 7]
```diff
@@ -1,37 +1,372 @@
...
+
+  **Step 5:**  If this is first time processing capsule, Set state machine to processing state which indicates capsule processing is initiated.
```
Comment (rprangar) [inline source/security/firmware-update.rst]: What do you mean by "If this is first time processing capsule"

File: source/security/firmware-update.rst [hunk 8]
```diff
@@ -1,37 +1,372 @@
...
+
+.. note:: Please refer to UEFI specification for more details about capsule header and data.
 
 Triggering Firmware Update 
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Triggering firmware update is a platform specific mechanism. The location of the firmware update capsule image is passed to Firmware update payload through configuration data.
+Triggering firmware update is a platform specific mechanism.
+
+|SPN| uses platform dependent register that can survive a reset to signal firmware update. Please refer to **Triggering Firmware Update** section of desired board page in **Supported Hardware** to find Sample implementation.
```
Comment (rprangar) [inline source/security/firmware-update.rst]: platform specific instead of platform dependent

Comment (raghavag) [reply source/security/firmware-update.rst]: Done

File: source/security/firmware-update.rst (hunk 6 above)
Comment (raghavag) [reply source/security/firmware-update.rst]: Changed it to Firmware Update Payload

File: source/security/firmware-update.rst (hunk 7 above)
Comment (raghavag) [reply source/security/firmware-update.rst]: Changed this line to "If current State Machine is set to Init, Set state machine to processing state which indicates capsule processing is initiated."

File: source/security/firmware-update.rst (hunk 5 above)
Comment (raghavag) [reply source/security/firmware-update.rst]: Done

File: source/security/firmware-update.rst (hunk 4 above)
Comment (raghavag) [reply source/security/firmware-update.rst]: Done

File: source/security/firmware-update.rst (hunk 1 above)
Comment (raghavag) [reply source/security/firmware-update.rst]: Done

**Pull request merged**